pkg/apis/workflow/v1alpha1/types.go

```diff
@@ -217,6 +217,11 @@ type Template struct {
 
 	// Tolerations to apply to workflow pods.
 	Tolerations []apiv1.Toleration `json:"tolerations,omitempty"`
+
+	// If specified, the pod will be dispatched by specified scheduler.
+	// If not specified, the pod will be dispatched by default scheduler.
+	// +optional
+	SchedulerName *string `json:"schedulerName,omitempty"`
```

Comment:
There's no need to make this a pointer, since a string with `omitempty` is better. It is also consistent with the pod.spec field:
```
	// If specified, the pod will be dispatched by specified scheduler.
	// If not specified, the pod will be dispatched by default scheduler.
	// +optional
	SchedulerName string `json:"schedulerName,omitempty" protobuf:"bytes,19,opt,name=schedulerName"`
```

Reply:
Also, can we duplicate this to the `workflow.spec`? I think it would be useful to have at both template and spec level. If both are specified, the template level schedulerName would take precedence.

Reply:
Actually, I started by adding a workflow level scheduler name in the first pr, and @alexmt suggested to make it container level. 
@alexmt how do you think about adding both?